lib/SIL/Utils/MemAccessUtils.cpp

```diff
@@ -877,9 +877,11 @@ SILValue swift::findOwnershipReferenceAggregate(SILValue ref) {
     if (auto *arg = dyn_cast<SILArgument>(root)) {
       if (auto *term = arg->getSingleTerminator()) {
         if (term->isTransformationTerminator()) {
-          assert(OwnershipForwardingTermInst::isa(term));
-          root = term->getOperand(0);
-          continue;
+          auto *ti = cast<OwnershipForwardingTermInst>(term);
+          if (ti->isDirectlyForwarding()) {
+            root = term->getOperand(0);
+            continue;
+          }
```

Comment:
Not critical in this PR, but it seems that 6 lines above should now have an assert:

```
  assert(cast<OwnershipForwardingTermInst>(root)->isDirectlyForwarding());
   root = root->getDefiningInstruction()->getOperand(0);
```